File: android/app/src/main/kotlin/xh/zero/flutter_demo/plugins/FlutterTextView.kt
package xh.zero.flutter_demo.plugins

import android.content.Context
import android.view.View
import android.widget.TextView
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.platform.PlatformView

class FlutterTextView(
        private val context: Context?,
        private val messenger: BinaryMessenger,
        id: Int
) : PlatformView, MethodChannel.MethodCallHandler {

    private val textView = TextView(context)

    private val methodChannel = MethodChannel(messenger, "xh.zero/textview_$id")

    init {
        methodChannel.setMethodCallHandler(this)
    }

    override fun getView(): View = textView

    override fun dispose() {
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        if (call.method == "setText") {
            val txt: String? = call.arguments as? String
            textView.text = txt;
            result.success(null)
        }
    }
}

File: ios/Runner/FlutterTextView.swift
//
//  TextView.swift
//  Runner
//
//  Created by xionghao on 2020/3/7.
//  Copyright © 2020 The Chromium Authors. All rights reserved.
//

import UIKit

class FlutterTextView: NSObject, FlutterPlatformView {
    private var channel: FlutterMethodChannel?
    private let label: UILabel!
    let frame: CGRect
    
    init(frame: CGRect, id: Int64, messenger: FlutterBinaryMessenger) {
        self.frame = frame
        print("frame \(frame)")
        self.label = UILabel(frame: frame)
        super.init()
       
        self.channel = FlutterMethodChannel(name: "xh.zero/textview_\(id)", binaryMessenger: messenger)
        self.channel?.setMethodCallHandler {[weak self] (call, result) in
            if (call.method == "setText") {
                let argument = call.arguments as? String
                self?.label.text = argument
            }
        }
    }
    
    func view() -> UIView {
        print("getView")
        return label
    }

}
